README.md

```diff
@@ -17,33 +17,54 @@ dev-dependencies:
 ### Basic options
 ```yaml
 license_generator:
-  fail_fast: true    #Errors are not ignored and the generator will fail with an error
+  fail_fast: true/false    #(not required) Errors are not ignored and the generator will fail with an error
+```
+
+### All options
+```yaml
+license_generator:
+  fail_fast: boolean    #(not required) Errors are not ignored and the generator will fail with an error -> Default = false
+  check_before_generate: boolean #(not required) this will check if your pubspec.yaml & pubspec.lock are in sync before generating the code -> Default = false
+  output_path: String #(not required) Override the default output path to generate the license file somewhere else -> Default = lib/util/license.dart
+  download_pub_dev_details: bool #(not required) This will download the pubdev details (homepage/repository) -> Default = false
+  pub_dev_base_url: String #(not required) This will override the default pubdev base url for downloading the pubdev details (homepage/repository) -> Default = https://pub.dev
+  licenses: #(not required) This will accept key value pairs (String/String) -> (package name/raw license url)
+    #license_generator: https://raw.githubusercontent.com/icapps/flutter-icapps-license/master/LICENSE (example)
+  extra_licenses: #(not required) this will allow you to add extra licenses that are not added in your pubspec.yaml
```

Comment:
I would use Optional, then you can see at a glance if it's required or not. Now I have to check where is required and is there a word in front of it

Reply:
fair point :) I will change that